File: sdk/ios/SsdidCore/Sources/SsdidCore/Api/PresentationApi.swift
import Foundation

/// Facade for OpenID4VP presentation operations.
public struct PresentationApi {
    private let handler: OpenId4VpHandler

    init(handler: OpenId4VpHandler) {
        self.handler = handler
    }

    public func processRequest(uri: String) async throws -> PresentationReviewResult {
        try await handler.processRequest(uri: uri)
    }

    public func submitPresentation(
        authRequest: AuthorizationRequest,
        matchResult: VpMatchResult,
        selectedClaims: [String],
        algorithm: String,
        signer: @Sendable (Data) -> Data
    ) async throws {
        try await handler.submitPresentation(
            authRequest: authRequest,
            matchResult: matchResult,
            selectedClaims: selectedClaims,
            algorithm: algorithm,
            signer: signer
        )
    }
}

File: android/sdk/ssdid-core/src/main/java/my/ssdid/sdk/api/PresentationApi.kt
package my.ssdid.sdk.api

import my.ssdid.sdk.domain.oid4vp.AuthorizationRequest
import my.ssdid.sdk.domain.oid4vp.MatchResult
import my.ssdid.sdk.domain.oid4vp.OpenId4VpHandler
import my.ssdid.sdk.domain.oid4vp.PresentationReviewResult

class PresentationApi internal constructor(private val handler: OpenId4VpHandler) {
    suspend fun processRequest(uri: String): Result<PresentationReviewResult> =
        handler.processRequest(uri)

    suspend fun submitPresentation(
        authRequest: AuthorizationRequest,
        matchResult: MatchResult,
        selectedClaims: List<String>,
        algorithm: String,
        signer: CredentialSigner
    ): Result<Unit> = handler.submitPresentation(
        authRequest, matchResult, selectedClaims, algorithm
    ) { data -> signer.sign(data) }
}
